frontend/src/components/navigation/FileSelector.tsx

```diff
@@ -0,0 +1,289 @@
+import React, { useState, useEffect } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import {
+	Button,
+	Box,
+	Modal,
+	List,
+	ListItem,
+	ListItemIcon,
+	ListItemText,
+	Typography,
+	IconButton,
+	Collapse,
+} from "@mui/material";
+import FolderIcon from "@mui/icons-material/Folder";
+import InsertDriveFileIcon from "@mui/icons-material/InsertDriveFile";
+import ExpandMoreIcon from "@mui/icons-material/ExpandMore";
+import ImageIcon from "@mui/icons-material/Image";
+import AudioIcon from "@mui/icons-material/Audiotrack";
+import VideoIcon from "@mui/icons-material/VideoFile";
+import TextIcon from "@mui/icons-material/Description";
+
+import { fetchDatasets } from "../../actions/dataset";
+import { V2 } from "../../openapi";
+
+// Define the file details
+interface FileDetails {
+	fileId: string;
+	fileName: string;
+}
+
+// Define the RecursiveComponent component with props type
+interface RecursiveComponentProps {
+	item: FSItem;
+	depth?: number;
+	onSelectFile: (fileId: string, fileName: string) => void;
+}
+
+// Define a type for items in the directory structure
+interface FSItem {
+	datasetId: string;
+	id?: string;
+	label: string;
+	children?: FSItem[] | undefined;
+	type: string; // A FSItem can be a folder or a file,
+	content_type?: string;
+}
+
+const RecursiveComponent: React.FC<RecursiveComponentProps> = ({
+	item,
+	depth = 0,
+	onSelectFile,
+}) => {
+	const [expanded, setExpanded] = useState(false);
+	const [children, setChildren] = useState<FSItem[] | undefined>(item.children);
+	const isFolder = item.type === "folder";
+
+	// Function to generate Icon based on item type
+	const getIcon = () => {
+		if (isFolder) {
+			return <FolderIcon />;
+		}
```

Comment:
If possible can we differentiate it with actual folder?
For dataset, we use this icon `import DatasetIcon from "@mui/icons-material/Dataset";`. We use this in the search page.

e.g. "test new dataset delete" is a dataset, and "folder renamed" is actually a folder.
<img width="378" alt="image" src="https://github.com/clowder-framework/clowder2/assets/13950475/d675e8d9-4006-40e5-8a2e-3bbf94e2e50f">